Find a functionally equivalent code:com.android.tools.layoutlib.create.AbstractClassAdapter.renameTypeAsType:COMMENT
Method Modifier: 
Comment:/**
 * Renames an object type, e.g. "Lcom.package.MyClass;" or an array type that has an
 * object element, e.g. "[Lcom.package.MyClass;".
 * This is like renameType() except that it returns a Type object.
 * If the type doesn't need to be renamed, returns the input type object.
 */

@@ -5,11 +5,11 @@
     if (type.getSort() == Type.OBJECT) {
         String in = type.getInternalName();
         String newIn = renameInternalType(in);
-        if (newIn != in) {
+        if (!newIn.equals(in)) {
             return Type.getType("L" + newIn + ";");
         }
     } else if (type.getSort() == Type.ARRAY) {

